testing/validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java

```diff
@@ -103,16 +105,27 @@ public void validate() throws Exception {
           }
 
           List<Metric> actualMetricList = Lists.newArrayList();
-          addMetrics(
-              CloudWatchService.SERVICE_DIMENSION,
-              serviceNames,
-              expectedMetricList,
-              actualMetricList);
-          addMetrics(
-              CloudWatchService.REMOTE_SERVICE_DIMENSION,
-              remoteServiceNames,
-              expectedMetricList,
-              actualMetricList);
+
+          // Add sets of dimesion filters to use for each query to CloudWatch
+          List<List<Pair<String, String>>> dimensionLists = Lists.newArrayList();
+          for (String serviceName : serviceNames) {
+            dimensionLists.add(
+                Arrays.asList(new Pair<>(CloudWatchService.SERVICE_DIMENSION, serviceName)));
+          }
+          for (String remoteServiceName : remoteServiceNames) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, remoteServiceName)));
+          }
```

Comment:
Is this new logic able to pull `[RemoteService, RemoteTarget]` without pulling `[<other stuff>, RemoteService, RemoteTarget]`? If yes, we can add back the other remote service values to `remoteServiceNames`. If not, the code changes in this PR will cause flaky tests in the long run.

This is due to the fact that when trying to pull `[RemoteService, RemoteTarget]` if you also pull other aggregations that have the same two attributes you will pull metrics from all test runs in the past three hours that match the filter you set. CW listMetrics API cannot return all of these metrics reliably and we've seen this cause failures before. Please ensure your solution accounts for this by running a few tests in a row and seeing if metrics with attributes containing the wrong test ID appear in the `actualMetricList`.